src/ecalc/libraries/libecalc/common/libecalc/input/mappers/create_references.py

```diff
@@ -56,12 +56,58 @@ def create_references(configuration: PyYamlYamlModel, resources: Resources) -> R
         fuel_data.get(EcalcYamlKeywords.name): FuelMapper.from_yaml_to_dto(fuel_data)
         for fuel_data in configuration.fuel_types
     }
+
+    consumers_installations = []
+    for installation in configuration.installations:
+        if installation.get(EcalcYamlKeywords.generator_sets) is not None:
+            consumers_installations.append(
+                [
+                    consumer
+                    for consumers in installation[EcalcYamlKeywords.generator_sets]
+                    for consumer in consumers[EcalcYamlKeywords.consumers]
+                ]
+            )
+
+        if installation.get(EcalcYamlKeywords.fuel_consumers) is not None:
+            consumers_installations.append(list(installation[EcalcYamlKeywords.fuel_consumers]))
+
+    check_multiple_energy_models(consumers_installations)
+
     return References(
         models=models,
         fuel_types=fuel_types,
     )
 
 
+def check_multiple_energy_models(consumers_installations: list):
```

Comment:
```suggestion
def check_multiple_energy_models(consumers_installations: List[..?]):
```
correct typing, "list" is only supported in 3.10+